ml-agents-envs/mlagents/envs/env_manager.py

```diff
@@ -0,0 +1,38 @@
+from abc import ABC, abstractmethod
+from typing import List, Dict, NamedTuple, Optional
+from mlagents.envs import AllBrainInfo, BrainParameters, Policy, ActionInfo
+
+
+class StepInfo(NamedTuple):
```

Comment:
Just an idea: instead of having 3 dictionaries all keyed by the brain name, what about something like
```
class BrainStepInfo(NamedTuple):
  previous_brain_info: Optional[BrainInfo]
  current_brain_info: Optional[BrainInfo]
  action_info: Optional[ActionInfo]

StepInfo = Dict[str, BrainStepInfo] # or make it a class/NamedTuple
```
Not sure if that would help clean anything up.

Reply:
This was essentially trying to minimize the changes elsewhere since a some parts of the codebase expect `AllBrainInfo`.  What do you think about leaving it for now and deprecating `AllBrainInfo` and splitting this into an `AgentStep` and `EnvironmentStep` in a following PR?

Reply:
Sure